src/AWS4AuthRequest.jl

```diff
@@ -34,6 +34,17 @@ function request(::Type{AWS4AuthLayer{Next}},
     return request(Next, url, req, body; kw...)
 end
 
+# Normalize whitespace to the form required in the canonical headers.
+# Note that the expected format for multiline headers seems not to be explicitly
+# documented, but Amazon provides a test case for it, so we'll match that behavior.
+function _normalize_ws(s::AbstractString)
+    if any(isequal('\n'), s)
+        join(map(_normalize_ws, split(s, '\n')), ',')
+    else
+        replace(strip(s), r"\s+" => " ")
+    end
```

Comment:
There seems to be some subtleties that are worth commenting about. So you want to replace each `\n` with a `,` and also remove all whitespace around the newlines. Any remaining contiguous whitespace is replaced with a single space.